### DISCUSSION TITLE
Serial.setTxBit()

### DISCUSSION THREAD

**Discussion by AlessGhitHub** [SpenceKonde/ATTinyCore#788] [Support / Q & A]
Does the Serial.setTxBit() command work on all core versions?

I'm making a program for ATtiny85 with Arduino IDE 1.8.13 and using ATtinyCore 1.5.2

I'm trying to use the Serial.setTxBit() function for the TX pin on ATtiny85 with ATtinyCore but the TX pin doesn't change, it always stays on PB0.

`void setup() {
 Serial.setTxBit(2);
   Serial.begin(9600);

  
  delay(100);
}

void loop() {

  if (Serial.available() > 0) {
    String receivedString = Serial.readString();  
    
  
    if (receivedString == "Hello") {
      // Envia a mensagem "World!!!" pela comunicação serial
      Serial.println("World!!!");
    }
  }
  
} `

**Comment by SpenceKonde** [comment]
No, it's it 2.0.0-dev only. The call may be there elsewheer but doesn't work.

**Comment by AlessGhitHub** [comment]
Ok so to change the TX only in core 2.0.0. I can not change in another version?

I will then install version 2.0.0-dev

**Reply by SpenceKonde** [reply]
Of course the catch with 2.0.0-dev is that it's unknown if the new implementation of the builtin softserial works at all...

**Reply by AlessGhitHub** [reply]
Where is the documentation for version 1.5.2?

**Reply by SpenceKonde** [reply]
Buried somewhere in github in the other branches, Quality is mostly pretty bad pre 2.0.0-dev as far as docs go, It doesn't really matter though, because you get the datasheet-effect: It documents the intended behavior of setTxBit. Since that feature was introduced, it in fact did not behave in a manner resembling what was documented or functionality more generally (as you no doubt observed). and didn't exist at all before that. It was reported after the 1.5,x maint cutoff that it had in fact never worked. So that was fixed, I know that calculation works on 2.0.0, but I don;t know if the rest of serial is still working after what I did to it next. Which was rather substantial.... because well - the old tinySoftSerial had a number of known bugs, some were potentially severe. and which were unknown to me before that expedition. So piecewise, tinySoftSerial was more or less rewritten to streamline the library and with limited success to improve performance in a desperate attempt to grasp for decent baud rates.... with less of the C and more of the inline ASM. In the process I added the alternate RX pin options on the x61 some such. but the routines to generate the timings were all done out from scratch, with the arithmetic errors (hopefully) corrected, but their viability within the relevant operating conditions is proven only theoretically, not with actual testing, and as such depends on the model and what the ideal timing is and of course how accurately you predict how long it will take to enter the ISR. That time should now be predictable and short (leaning more heavily on inline ASM, because these are things that were *really* meant to be written in .S files, but I have yet to grow a pair and synthesize my own .S, and inline asm is equivalent, just uglier mostly.

I may not be as clever as I think I am *(there is an audible din of heckling - he continues undeterred)* so there may still be problems of various sorts with the builtin software serial port, which is exactly the sort of thing I need help from people to test.

**Reply by AlessGhitHub** [reply]
Summarizing your work is formidable, what would those who want to use an ATiny without your effort and dedication.

Congratulations
